docs: add README language selection links and Chinese docs link

dowithless opened the pull request:
Added language selection links in README for easier access to translations: German, Spanish, French, Japanese, Korean, Portuguese, Russian, and Chinese.

# Pull Request

**NautilusTrader prioritizes correctness and reliability, please follow existing patterns for validation and testing.**

- [ ] I have reviewed the `CONTRIBUTING.md` and followed the established practices

## Summary

<!-- Provide a brief description of *what* changed, *why* it was changed, and the impact on the system or users (2–3 sentences). -->

## Related Issues/PRs

<!-- List any related GitHub issues or PRs (e.g., `Closes #123`, `Related to #456`). -->

## Type of change

<!-- Select all that apply. -->

- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (impacts existing behavior)
- [x] Documentation update
- [ ] Maintenance / chore

## Breaking change details (if applicable)

<!-- If this is a breaking change, describe the impact and any migration steps required for users or developers. -->

## Documentation

- [ ] Documentation changes follow the style guide (`docs/developer_guide/docs.md`)

## Release notes

- [ ] I added a concise entry to `RELEASES.md` that follows the existing conventions (when applicable)

## Testing

**Ensure new or changed logic is covered by tests.**

- [ ] Affected code paths are already covered by the test suite
- [ ] I added/updated tests to cover new or changed logic

<!-- Briefly describe how the changes were tested (e.g., unit tests in `tests/unit/test_file.py`, or *additional* manual testing). -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/nautechsystems/nautilus_trader?pullRequest=2902) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/nautechsystems/nautilus_trader?pullRequest=2902) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/nautechsystems/nautilus_trader?pullRequest=2902) it.</sub>

dowithless commented:
> Hi @dowithless
> 
> Thanks for the PR, this is a good idea.
> 
> Let's move this down under the [Community](https://github.com/nautechsystems/nautilus_trader?tab=readme-ov-file#community) section:
> 
> ```
> ## Community
> 
> Join our community of users and contributors on [Discord](https://discord.gg/NautilusTrader) to chat
> and stay up-to-date with the latest announcements and features of NautilusTrader. Whether you're a
> developer looking to contribute or just want to learn more about the platform, all are welcome on our Discord server.
> 
> ### Translations
> These are automatically translated README versions hosted by a third party (unofficial, accuracy may vary).
> 
> <!-- Translations will automatically update with the README. -->
> [Deutsch](https://zdoc.app/de/nautechsystems/nautilus_trader) | 
> [Español](https://zdoc.app/es/nautechsystems/nautilus_trader) | 
> [Français](https://zdoc.app/fr/nautechsystems/nautilus_trader) | 
> [日本語](https://zdoc.app/ja/nautechsystems/nautilus_trader) | 
> [한국어](https://zdoc.app/ko/nautechsystems/nautilus_trader) | 
> [Português](https://zdoc.app/pt/nautechsystems/nautilus_trader) | 
> [Русский](https://zdoc.app/ru/nautechsystems/nautilus_trader) | 
> [中文](https://zdoc.app/zh/nautechsystems/nautilus_trader)
> ```

Thanks for the feedback! I've updated the PR and moved it under the Community section as suggested. 👍
Please review again.

dowithless commented:
> @dowithless
> 
> After considering this further, I think it’s of limited value to translate only the README, and it may add unnecessary clutter. Users who are motivated can readily find translations themselves.
> 
> Thanks again for the contribution and for your understanding!

Thanks for the feedback!

Just to clarify — [zdoc.app](https://zdoc.app/) isn’t limited to translating only the README, it can provide full documentation site translations.

I noticed the docs site here: https://nautilustrader.io/docs/latest/

It looks like it might be built with Docusaurus. The nautilus_trader repo has a docs folder, but I didn’t see any Docusaurus config (e.g. docusaurus.config.js). My guess is that this repo just stores the source docs, and they’re rendered elsewhere (maybe in a separate site repo or with another tool).

If I could learn how the docs site is actually rendered, I’d be happy to explore translating the full site.

For reference, here are some existing translated docs powered by zdoc:

https://www.zdoc.app/zh/Significant-Gravitas/AutoGPT

https://www.zdoc.app/zh/Shubhamsaboo/awesome-llm-apps

https://www.zdoc.app/zh/menloresearch/jan

On those pages, you’ll notice a link to the translated docs/site conveniently shown on the right-hand side.

dowithless commented:
> I think that contributors would prefer to focus on developing the library further and developing the documentation in English rather than maintaining the documentation in multiple languages. There are not that many contributors so what we choose to do should be chosen carefully. Most libraries related to trading are only in English which is the standard language of programming and finance.

I understand your point.
According to analysis from OSS Insight, the majority of stargazers for nautilus_trader are from China. Providing documentation in Chinese could therefore be valuable for the project.

Also, the translations are maintained by zdoc, so this won’t add any extra maintenance burden for the core nautilus_trader developers.

<img width="1957" height="972" alt="image" src="https://github.com/user-attachments/assets/19fb7881-7166-4de3-9867-0982298a3b62" />
